app/models/package_manager/go.rb

```diff
@@ -21,32 +23,71 @@ class Go < Base
 
 
     def self.package_link(project, version = nil)
-      "http://go-search.org/view?id=#{project.name}"
+      "https://pkg.go.dev/#{project.name}#{"@#{version}" if version}"
     end
 
     def self.documentation_url(name, version = nil)
-      "http://godoc.org/#{name}"
+      "https://pkg.go.dev/#{name}#{"@#{version}" if version}?tab=doc"
     end
 
     def self.install_instructions(project, version = nil)
       "go get #{project.name}"
     end
 
     def self.project_names
-      get("http://go-search.org/api?action=packages")
+      # Currently the index only shows the last <=2000 modules from the date given. (https://proxy.golang.org/)
+      project_window = 1.day.ago.strftime("%FT%TZ")
+      get_raw("https://index.golang.org/index?since=#{project_window}&limit=2000")
+        .lines
+        .map { |line| JSON.parse(line)["Path"] }
     end
 
     def self.project(name)
-      get("http://go-search.org/api?action=package&id=#{name}")
+      if pkg_html = get_html("https://pkg.go.dev/#{name}?tab=doc")
+        go_module, _latest_version = pkg_html.css('a[data-test-id="DetailsHeader-infoLabelModule"]').first&.text&.split('@', 2)
+        go_module_html = get_html("https://pkg.go.dev/mod/#{go_module}")
+        # NB this requires a quick request for each version, but we could alternatively fetch from
+        # https://pkg.go.dev/mod/#{go_module}?tab=versions and the '.Versions-commitTime', selector,
+        # but the dates are a combination of date-only or natural language (e.g. '1 day ago')
+        versions = get_raw("http://proxy.golang.org/#{go_module}/@v/list")
+          &.lines
+          &.map(&:strip)
+          &.reject(&:blank?)
+          &.map do |v|
+            info = get("http://proxy.golang.org/#{go_module}/@v/#{v}.info")
+            {
+              number: info["Version"],
+              published_at: info["Time"].presence && Time.parse(info["Time"])
+            }
+          end
```

Comment:
This is the only questionable part -- it's a choice between two options:

1) this, where we make a superfast request for each version to get it's actual published_at
2) or, make a single request to the pkg.go.dev page, and get each version with a Date-only-version of its published_at

🤷‍♂️

Reply:
I like the approach here of doing a request per version. This is a background job and so would rather get the full and correct info

Reply:
@katzj I just switched this to use the `pkg.go.dev` scraping instead, because I noticed `pkg.go.dev` will list files that aren't technically Go Modules, e.g. https://pkg.go.dev/github.com/BurntSushi/toml. Whereas the Go proxy url fails 😕 :

``` sh
 $ curl http://proxy.golang.org/github.com/BurntSushi/toml/@v/list
bad request: invalid escaped module path "github.com/BurntSushi/toml"
```

Seem ok?